app/services/katello/registration_manager.rb

```diff
@@ -83,6 +83,10 @@ def validate_hosts(hosts, organization, host_name, host_uuid, host_uuid_overridd
                              org_name: organization.name, host_name: host_name)
         end
 
+        if !host.build && Setting[:host_re_register_build_only]
```

Comment:
This new block should come after the `  if host.name == host_name` line just below so that `host` is defined (but before the DMI UUID checking)

Reply:
Oops.. pushed a fix.